updater/aws.go

```diff
@@ -119,27 +150,35 @@ func (u *updater) filterAvailableUpdates(bottlerocketInstances []instance) ([]in
 		instances = append(instances, inst.instanceID)
 	}
 
-	commandID, err := u.sendCommand(instances, u.checkDocument)
-	if err != nil {
-		return nil, err
-	}
-
 	candidates := make([]instance, 0)
-	for _, inst := range bottlerocketInstances {
-		commandOutput, err := u.getCommandResult(commandID, inst.instanceID)
-		if err != nil {
-			return nil, err
-		}
-		output, err := parseCommandOutput(commandOutput)
+	_, err := eachPage(len(instances), ssmPageSize, func(start, stop int) error {
+		commandID, err := u.sendCommand(instances[start:stop], u.checkDocument)
 		if err != nil {
-			// not a fatal error, we can continue checking other instances.
-			log.Printf("Failed to parse command output %q: %v", string(commandOutput), err)
-			continue
+			// errors here are considered non-fatal.
+			log.Printf("Unable to send document %s: %v", u.checkDocument, err)
+			return nil
 		}
-		if output.UpdateState == updateStateAvailable || output.UpdateState == updateStateReady {
-			inst.bottlerocketVersion = output.ActivePartition.Image.Version
-			candidates = append(candidates, inst)
+		for _, inst := range bottlerocketInstances[start:stop] {
+			commandOutput, err := u.getCommandResult(commandID, inst.instanceID)
+			if err != nil {
+				// errors here are considered non-fatal
+				log.Printf("Unable to retrieve output for command ID %s, documet %s: %v", commandID, u.checkDocument, err)
+				continue
+			}
+			output, err := parseCommandOutput(commandOutput)
+			if err != nil {
+				log.Printf("Failed to parse command output %s: %v", string(commandOutput), err)
+				continue
```

Comment:
instance details would help in debugging and %q for quoting the command output.
```suggestion
				log.Printf("Failed to parse command output %q for instance %q: %v", string(commandOutput), inst, err)
				continue
```